operator/duties/scheduler_test.go

```diff
@@ -17,35 +17,79 @@ import (
 	"github.com/bloxapp/ssv/logging"
 	"github.com/bloxapp/ssv/networkconfig"
 	"github.com/bloxapp/ssv/operator/duties/mocks"
+	"github.com/bloxapp/ssv/operator/slot_ticker"
 	mockslotticker "github.com/bloxapp/ssv/operator/slot_ticker/mocks"
 	mocknetwork "github.com/bloxapp/ssv/protocol/v2/blockchain/beacon/mocks"
 )
 
+type MockSlotTicker interface {
+	Next() <-chan time.Time
+	Slot() phase0.Slot
+	Subscribe() chan phase0.Slot
+}
+
 type mockSlotTicker struct {
-	event.Feed
+	slotChan chan phase0.Slot
+	timeChan chan time.Time
+	slot     phase0.Slot
+	mu       sync.Mutex // Added Mutex for synchronization
+}
+
+func NewMockSlotTicker() MockSlotTicker {
+	ticker := &mockSlotTicker{
+		slotChan: make(chan phase0.Slot),
+		timeChan: make(chan time.Time),
+	}
+	ticker.start()
+	return ticker
+}
+
+func (m *mockSlotTicker) start() {
+	go func() {
+		for slot := range m.slotChan {
+			m.mu.Lock() // Locking here
+			m.slot = slot
+			m.mu.Unlock() // Unlocking here
+			m.timeChan <- time.Now()
+		}
+	}()
+}
+
+func (m *mockSlotTicker) Next() <-chan time.Time {
+	return m.timeChan
+}
+
+func (m *mockSlotTicker) Slot() phase0.Slot {
+	m.mu.Lock()         // Locking here
+	defer m.mu.Unlock() // Unlocking with defer
+	return m.slot
+}
+
+func (m *mockSlotTicker) Subscribe() chan phase0.Slot {
+	return m.slotChan
 }
 
-func (m *mockSlotTicker) Subscribe(subscriber chan phase0.Slot) event.Subscription {
-	return m.Feed.Subscribe(subscriber)
+type mockSlotTickerService struct {
+	event.Feed
 }
 
 func setupSchedulerAndMocks(t *testing.T, handler dutyHandler, currentSlot *SlotValue) (
 	*Scheduler,
 	*zap.Logger,
-	*mockSlotTicker,
+	*mockSlotTickerService,
 	time.Duration,
 	context.CancelFunc,
 	*pool.ContextPool,
 ) {
 	ctrl := gomock.NewController(t)
-	timeout := 100 * time.Millisecond
+	timeout := 200 * time.Millisecond
```

Comment:
Why do we need this change? Has the timing behavior changed? If so, I think we need to comment it